src/com/goide/inspections/GoStructInitializationInspection.java

```diff
@@ -32,17 +41,87 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
 
-        if (PsiTreeUtil.getParentOfType(o, GoReturnStatement.class, GoShortVarDeclaration.class, GoAssignmentStatement.class) == null) return;
+        if (PsiTreeUtil.getParentOfType(o, GoReturnStatement.class, GoShortVarDeclaration.class, GoAssignmentStatement.class) == null) {
+          return;
+        }
+
+        if (!(o.getContainingFile() instanceof GoFile)) return;
+        String myPackage = ((GoFile)o.getContainingFile()).getPackageName();
+
+        List<GoFieldDeclaration> fields;
+        GoType type = o.getType();
+        if (type instanceof GoStructType) {
+          if (!canProcess(myPackage, (GoStructType)type)) return;
+          fields = ((GoStructType)type).getFieldDeclarationList();
+          if (hasAnonymousStructField(fields)) return;
+          processLiteralValue(holder, fields, o.getLiteralValue());
+          return;
+        }
 
         GoTypeReferenceExpression ref = o.getTypeReferenceExpression();
-        if (o.getType() instanceof GoStructType || ref != null && ref.resolveType() instanceof GoSpecType) {
+        if (ref == null || ref.resolveType() == null) return;
+        GoType refType = ref.resolveType().getUnderlyingType();
+        if (refType instanceof GoStructType) {
+          if (!canProcess(myPackage, (GoStructType)refType)) return;
+          fields = ((GoStructType)refType).getFieldDeclarationList();
+          if (hasAnonymousStructField(fields)) return;
+          processLiteralValue(holder, fields, o.getLiteralValue());
+        }
+        else if (refType instanceof GoArrayOrSliceType) {
+          type = ((GoArrayOrSliceType)refType).getType();
+          if (type == null || !(type.getUnderlyingType() instanceof GoStructType)) return;
+          if (!canProcess(myPackage, (GoStructType)type.getUnderlyingType())) return;
+          fields = ((GoStructType)type.getUnderlyingType()).getFieldDeclarationList();
+          if (hasAnonymousStructField(fields)) return;
           for (GoElement element : o.getLiteralValue().getElementList()) {
-            if (element.getKey() == null) {
-              holder.registerProblem(element, "Unnamed field initialization", ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
-            }
+            GoValue value = element.getValue();
+            if (value == null || value.getExpression() != null || value.getLiteralValue() == null) continue;
+            processLiteralValue(holder, fields, value.getLiteralValue());
           }
         }
       }
     };
   }
+
+  @Override
+  public JComponent createOptionsPanel() {
+    return new SingleCheckboxOptionsPanel("Report for local type definitions as well", this, "reportImportedStructs");
+  }
+
+  private boolean canProcess(String myPackage, @NotNull GoStructType type) {
+    if (!(type.getContainingFile() instanceof GoFile)) return false;
+    String typePackage = ((GoFile)type.getContainingFile()).getPackageName();
+    return reportImportedStructs || myPackage == typePackage;
```

Comment:
comparing package's names is not enough to say that symbols are located in the same package